public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,885 @@
+
+:marked
+  Click the "_Add a Secret Lair_" button. 
+  A new address section appears. Well done!
+
+  ### Remove a lair
+
+  This example can _add_ addresses but it can't _remove_ them.
+  For extra credit, write a `removeLair` method and wire it to a button on the repeating address HTML.
+
+.l-main-section
+a#watch-control
+:marked
+  ## Watch control changes
+
+  Angular calls `ngOnChanges` when the user picks a hero in the parent `HeroListComponent`.
+  Picking a hero changes the `HeroDetailComponent.hero` input property.
+  
+  Angular does _not_ call `ngOnChanges` when the user modifies the hero's _name_ or _secret lairs_.
+  Fortunately, you can learn about such changes by subscribing to one of the form control properties 
+  that raises a change event.
+  
+  These are properties, such as `valueChanges`, that return an RxJS `Observable`.
+  You don't need to know much about RxJS `Observable` to watch for a form control value change.
+
+  Add the following method to watch for changes to the value of the _name_ `FormControl`.
++makeExample('reactive-forms/ts/app/hero-detail.component.ts', 'watch-name','app/hero-detail.component.ts (watchName)')(format=".")
+
+:marked
+  Call it in the constructor, after creating the form.
++makeExample('reactive-forms/ts/app/hero-detail.component.ts', 'ctor')(format=".")
+
+:marked
+  The `watchName` method pushes name-change values into a `nameChangeLog` array.
+  Display that array at the bottom of the component template with this `*ngFor` binding:
++makeExample('reactive-forms/ts/app/hero-detail.component.html', 'name-change-log','app/hero-detail.component.html (Name change log)')(format=".")
+
+:marked
+  Return to the browser, select a hero (e.g, "Magneta"), and start typing in the _name_ input box.
+  You should see a new name in the log after each keystroke.
+
+  ### When to use it
+  
+  An interpolation binding is the easier way to _display_ a name change.
+  Subscribing to an observable form control property is handy for triggering 
+  application logic _within_ the component class.
+
+.l-main-section
+a#save
+:marked
+  ## Save form data
+
+  The `HeroDetailComponent` captures user input but it doesn't do anything with it.
+  In a real app, you'd probably save those hero changes.
+  In a real app, you'd also be able to revert unsaved changes and resume editing.
+  After you implement both features in this section, the form will look like this:
+
+figure.image-display
+  img(src="/resources/images/devguide/reactive-forms/save-revert-buttons.png" width="389px" alt="Form with save & revert buttons")
+
+:marked
+  ### Save
+  In this sample application, the `HeroDetailComponent` cannot perform the save operation. 
+  The component delegates that chore to its parent `HeroListComponent`.
+  
+  When the user clicks the _Save_ button, the `HeroDetailComponent` raises a save event on its `save` output property.
```

Comment:
I think we should leave this section out. Demonstrating Output and component interaction isn't really the point of the forms guide, so it's adding complexity and bloat without adding value.  

That said, I like that it explains ngSubmit. We should keep that event, but the method could just be a placeholder method with a comment  (// here is where we'd post to server) should be sufficeint.